jubilant/_juju.py

```diff
@@ -305,6 +305,41 @@ def config(
 
         self.cli(*args)
 
+    def consume(
+        self,
+        model_and_app: str,
+        alias: str | None = None,
+        *,
+        controller: str | None = None,
+        owner: str | None = None,
+    ) -> None:
+        """Add a remote offer to the model.
+
+        Examples::
+
+            juju.consume('othermodel.mysql', 'sql')
+            juju.consume('othermodel.mysql', controller='ctrl2', owner='admin')
+
+        Args:
+            model_and_app: Dotted application and model name to offer endpoints for, for example
+                ``othermodel.mysql``.
+            alias: A local alias for the offer, for use with :meth:`integrate`. Defaults to the
+                application name.
+            controller: Remote offer's controller. Defaults to the current controller.
+            owner: Remote model's owner. Defaults to the user that is currently logged in to the
+                controller providing the offer.
+        """
+        offer_path = model_and_app
+        if owner is not None:
+            offer_path = owner + '/' + offer_path
+        if controller is not None:
+            offer_path = controller + ':' + offer_path
```

Comment:
We could handle empty strings gracefully (ignoring them and treating them as the option not being specified) by testing for truthiness, avoiding sending a (presumably?) malformed string to Juju at runtime. On the other hand, maybe this has the potential to silently hide user errors.
```suggestion
        if owner:
            offer_path = owner + '/' + offer_path
        if controller:
            offer_path = controller + ':' + offer_path
```